[14.0] Add connector_importer_product

mmequignon opened the pull request:
Base product importer.

mmequignon committed:
Add connector_importer_product

simahawk commented on connector_importer_product/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1 @@
+Generic module to import product throught SFTP
```
is not for SFTP only

sebalix commented on connector_importer_product/data/import_recordset.xml:
```diff
@@ -0,0 +1,54 @@
+  <record id="import_recordset_product_category" model="import.recordset">
+    <field name="sequence" eval="20" />
+    <field name="backend_id" ref="import_backend" />
+    <field name="source_model">import.source.csv.sftp</field>
```
Should not be tied to SFTP only
```suggestion
    <field name="source_model">import.source.csv.sftp</field>
```

sebalix commented on connector_importer_product/data/import_source.xml:
```diff
@@ -0,0 +1,60 @@
+    <field name="csv_delimiter">,</field>
+    <field name="csv_encoding">utf-8</field>
+    <field name="storage_id" ref="storage_backend_importer" />
+    <field name="move_file_after_import" eval="1" />
```
This is also a configuration brought by SFTP, to remove I guess

TDu commented on connector_importer_product/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+        "data/import_type.xml",
+        "data/import_source.xml",
+        "data/import_recordset.xml",
+        # Demo
```
```suggestion
```

TDu commented on connector_importer_product/components/common.py:
```diff
@@ -0,0 +1,22 @@
+    _break_on_error = True
+
+    def make_translation_key(self, key, lang):
+        """Overridden to handle 'name_fr' and 'name_de' fields."""
```
Need to be reformulated it is not specific to fr or de, explain its use.

TDu commented on connector_importer_product/components/product_attribute.py:
```diff
@@ -0,0 +1,53 @@
+from ..utils import sanitize_external_id
+
+
+class ProductAttributeRecordImporter(Component):
```
```suggestion
class ProductAttributeImporter(Component):
```

TDu commented on connector_importer_product/components/product_attribute_value.py:
```diff
@@ -0,0 +1,51 @@
+from ..utils import sanitize_external_id
+
+
+class ProductAttributeValueRecordImporter(Component):
```
```suggestion
class ProductAttributeValueImporter(Component):
```

TDu commented on connector_importer_product/components/product_category.py:
```diff
@@ -0,0 +1,70 @@
+    return modifier
+
+
+class ProductCategoryRecordImporter(Component):
```
```suggestion
class ProductCategoryImporter(Component):
```

TDu commented on connector_importer_product/components/product_packaging.py:
```diff
@@ -0,0 +1,19 @@
+from ..utils import sanitize_external_id
+
+
+class ProductPackagingRecordImporter(Component):
```
```suggestion
class ProductPackagingImporter(Component):
```

TDu reviewed (commented):
About my comment of the class name I just stopped after a while of reviewing, I means it is not that important

simahawk commented on connector_importer_product/data/import_backend.xml:
```diff
@@ -0,0 +1,11 @@
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+  <record id="import_backend" model="import.backend">
```
you should trash all these records

simahawk commented on connector_importer_product/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    "license": "AGPL-3",
+    "depends": [
+        # oca
+        "connector_importer_source_sftp",
```
this should not be a dependency

simahawk commented on connector_importer_product/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    ],
+    "data": [
+        # storage_backend
+        "data/storage_backend.xml",
```
same

simahawk committed:
connector_importer: add bool converter

simahawk committed:
connector_importer: manage break flag via options

You can now configure an importer like this:

  options:
    importer:
      break_on_error: true

to force it to break.

simahawk committed:
connector_importer: transl key config via options

You can now configure the lang key at import type level
by settings these keys on the importer options:

  options:
    importer:
      translation_key_sep: _
      translation_use_regional_lang: false

simahawk commented:
@sebalix @mmequignon I've pushed some cleanup/fixes/improvements.

mmequignon reviewed (commented):
@simahawk I cannot approve my own pr, but lg.
Thx!

simahawk commented:
I switched tests to GH actions on #89.Rebased and squashed here.

simahawk committed:
connector_importer: fix import_record config from job

When importer config is loaded from the DB via job arguments
you don't have a DotDict anymore.

It should be fixed by passing the importer_type_id IMO but for now is ok.

simahawk committed:
connector_importer: fix rendering of docs

Importers taken to generate docs must be initialized in the same way
as whem they are used for importing records.
This way 'options' and 'context' are propagated properly.

simahawk committed:
connector_importer_product: mapper fixes

simahawk commented:
/ocabot merge minor

simahawk commented:
/ocabot merge minor
